addon/components/base/bs-collapse.js

```diff
@@ -123,13 +124,14 @@ export default class Collapse extends Component {
    */
   transitionDuration = 350;
 
+  collapseSize = null;
+
   setCollapseSize(size) {
     let dimension = this.get('collapseDimension');
 
     assert(`collapseDimension must be either "width" or "height". ${dimension} given.`, ["width", "height"].indexOf(dimension) !== -1);
 
-    this.element.style.width = dimension === 'width' && size ? `${size}px` : '';
-    this.element.style.height = dimension === 'height' && size ? `${size}px` : '';
+    this.set('collapseSize', size);
   }
```

Comment:
Maybe `collapseSize` and `setCollapseSize` should be refactored to native getter and setter for readability?

```js
@computed('_collapseSize')
get collapseSize() {
  return this._collapseSize;
}
set collapseSize(size) {
  let dimension = this.get('collapseDimension'); 

  assert(`collapseDimension must be either "width" or "height". ${dimension} given.`, ["width", "height"].indexOf(dimension) !== -1);

  this.set('_collapseSize', size);
}

_collapseSize: null;
```

Will be even easier to read as soon as we introduce tracked properties.

Reply:
Tbh there is a ton of code that needs to be refactored, especially in that class (e.g. observers). But I intentionally wanted to make the changes as small as possible, only just as much to enable the main goal here, which is moving to outerHTML semantics. But again yes, the code is ages old and definitely needs some love! 😍